sklearn/exceptions.py

```diff
@@ -154,3 +155,12 @@ class UndefinedMetricWarning(UserWarning):
     .. versionchanged:: 0.18
        Moved from sklearn.base.
     """
+
+
+class SklearnDeprecationWarning(DeprecationWarning):
```

Comment:
I'm fine with any side of this tradeoff - it is either 2 lines of code in `__init__` or less strict backwards compatibility for filtering of deprecation warnings. Backwards compatibility looked a bit better to me, and this is what @amueller indirectly suggested in a comment, so I went with it, but I'd be happy to change that if there is a consensus backwards compatibility here is not needed.